README.md

```diff
@@ -2,58 +2,162 @@
 
-## GitHub Actions & CI
+## Signing and Notarizing on macOS
 
-The scripts contained in `github/scripts` can be used to build and package the plugin and take care of setting up obs-studio as well as its own dependencies. A default workflow for GitHub Actions is also provided and will use these scripts.
+Plugins released for macOS should be codesigned and notarized with a valid Apple Developer ID for best user experience. To set this up, the private and personal key of a **paid Apple Developer ID** need to be downloaded from the Apple Developer portal:
 
-### Retrieving build artifacts
+* On your Apple Developer dashboard, go to "Certificates, IDs & Profiles" and create two signing certificates:
+    * One of the "Developer ID Application" type. It will be used to sign the plugin's binaries
+    * One of the "Developer ID Installer" type. It will be used to sign the plugin's installer
 
-Each build produces installers and packages that you can use for testing and releases. These artifacts can be found on the action result page via the "Actions" tab in your GitHub repository.
+The developer certificate will usually carry a name similar in form to
 
-#### Building a Release
+`Developer ID Application: <FIRSTNAME> <LASTNAME> (<LETTERS_AND_NUMBERS>)`
 
-Simply create and push a tag and GitHub Actions will run the pipeline in Release Mode. This mode uses the tag as its version number instead of the git ref in normal mode.
+This entire string should be specified as `CODESIGN_IDENTITY`, the `LETTERS_AND_NUMBERS` part as `CODESIGN_TEAM` to CMake to set up codesigning properly.
 
-### Packaging on Linux
+### GitHub Actions Set Up
 
-The install step results in different directory structures depending on the value of `LINUX_PORTABLE` - "OFF" will organize outputs to be placed in the system root, such as `/usr/`, and "ON" will organize outputs for portable installations in the user's home directory. If you are packaging for a Linux distribution, you probably want to set `-DLINUX_PORTABLE=OFF`.
+To use code signing on GitHub Actions, the certificate and associated information need to be set up as _repository secrets_ in the GitHub repository's settings:
 
-### Signing and Notarizing on macOS
+* Using the Keychain app on macOS, export these your certificates (Application and Installer) public _and_ private keys into a single .p12 file **protected with a strong password**
+* Encode the .p12 file into its base64 representation by running `base64 <NAME_OF_YOUR_P12_FILE>`
+* Add your developer certificates' names (see above):
+    * `MACOS_SIGNING_APPLICATION_IDENTITY`: Name of the "Developer ID Application" signing certificate
+    *  `MACOS_SIGNING_INSTALLER_IDENTITY`: Name of "Developer ID Installer" signing certificate
+* `MACOS_SIGNING_CERT`: The base64 encoded `.p12` file
+* `MACOS_SIGNING_CERT_PASSWORD`: Password used to generate the .p12 certificate
```

Comment:
These seem to require additional information about adding them as secrets. They seem a bit detached otherwise.

Reply:
The block is explicitly about necessary repository secrets and the first two bullets explain how to generate the file (which requires choosing a password). I don't see what other information might be necessary.

Reply:
![image](https://github.com/obsproject/obs-plugintemplate/assets/624931/9d9ac352-2e04-4eae-80e2-38a1c80c775a)

The bulleted list items are just a bit disjointed to me. The list is prefaced with:
> To use code signing on GitHub Actions, the certificate and associated information need to be set up as repository secrets in the GitHub repository's settings:

So in that context, these two items make sense. However, they are separated from that preface by intermediate instructions, so I did not immediately connect these two lines to the preface, but instead connected them to the sub-list immediately above.

Reply:
Looks much better. Thanks!